### DISCUSSION TITLE
Is there a periodic servicewindow where d365fo services not available ?

### DISCUSSION THREAD

**Discussion by Myhlendorph** [d365collaborative/d365fo.tools#547] [General]
Hi 
Trying to install this morning but I get the following
It use to work

PackageManagement\Install-Package : No match was found for the specified search criteria and module name
'd365fo.tools'. Try Get-PSRepository to see all available registered module repositories.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1.0.0.1\PSModule.psm1:1772 char:21
+ ...          $null = PackageManagement\Install-Package @PSBoundParameters
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Microsoft.Power....InstallPackage:InstallPackage) [Install-Package], Ex
   ception
    + FullyQualifiedErrorId : NoMatchFoundForCriteria,Microsoft.PowerShell.PackageManagement.Cmdlets.InstallPackage

Import-Module : The specified module 'd365fo.tools' was not loaded because no valid module file was found in any
module directory.

**Comment by Splaxi** [comment]
The Install-Module, which is a standard module in powershell, supports multiple repositories - think liket nuget.

The default repository is the www.powershellgallery.com and out module lives its live here: https://www.powershellgallery.com/packages/d365fo.tools/0.6.58

So whenever you run Install-Module d365fo.tools, you are depending on your internet connection, to be able to reach www.powershellgallery.com and hits its API, that will give you the latest version of the module.

Without knowing the finer details of your environment or how you use it, you should know that 1-2 years back there was a change, that enforced TLS1.2 on the www.powershellgallery.com. This impacts PowerShell 5.1, the default installation on all Windows 10 / Servers 2016, because it depends on a registry settings, which isn't defaulting to TLS1.2. So you either need to set the tls1.2 before running the install-module or you need to locate the registry setting and make sure it is configured correctly.

Or you could simply have had an connectivity issue, www.powershellgallery.com could have had a minor incident - and the above explanation totally useless for you, but now you know 😉

Try again and let me know if you keep facing the same issue.

Example on how to ensure TLS1.2:
```powershell
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
Install-Module d365fo.tools
```

**Reply by Splaxi** [reply]
Okay. I've learned that the team behind the powershell gallery didn't renew their SSL certificate in time, which caused a full stop of the service. 

So there you have it 🤷‍♂️

**Reply by Myhlendorph** [reply]
I found it to work a couple of hours later.
Thank you for the insights
Best regards
